src/mpsc/errors.rs

```diff
@@ -88,6 +108,72 @@ impl<T> fmt::Display for Closed<T> {
 #[cfg(feature = "std")]
 impl<T> std::error::Error for Closed<T> {}
 
+// === impl SendTimeoutError ===
+
+impl SendTimeoutError {
+    #[cfg(feature = "std")]
+    pub(crate) fn with_value<T>(self, value: T) -> SendTimeoutError<T> {
+        match self {
+            Self::Timeout(()) => SendTimeoutError::Timeout(value),
+            Self::Closed(()) => SendTimeoutError::Closed(value),
+        }
+    }
+}
+
+impl<T> SendTimeoutError<T> {
+    /// Returns `true` if this error was returned because the channel is still
+    /// full after the timeout has elapsed.
+    pub fn is_timeout(&self) -> bool {
+        matches!(self, Self::Timeout(_))
+    }
+
+    /// Returns `true` if this error was returned because the channel has closed
+    /// (e.g. the [`Receiver`] end has been dropped).
+    ///
+    /// If this returns `true`, no future [`try_send`] or [`send`] operation on
+    /// this channel will succeed.
+    ///
+    /// [`Receiver`]: super::Receiver
+    /// [`try_send`]: super::Sender::try_send
+    /// [`send`]: super::Sender::send
+    /// [`Receiver`]: super::Receiver
```

Comment:
since this error is only returned by the blocking channel, let's add:
```suggestion
    /// [`Receiver`]: super::blocking::Receiver
    /// [`try_send`]: super::blocking::Sender::try_send
    /// [`send`]: super::blocking::Sender::send
    /// [`Receiver`]: super::blocking::Receiver
```